### ISSUE TITLE
Category url keys that end in slash breaks layered navigation with 2.3.3

### ISSUE THREAD

**Issue by wibbel** [magento/magento2#25447]
### Preconditions (*)
When the magento configuration specifies a trailing slash as the Category URL Suffix, the layered navigation breaks. For example category-url-key/

1. Magento version 2.3.3
2. No customizations for the store, Luma + Sample Data
3. Modify the configuration to use category url suffix as a slash rather than .html (stores->configuration->catalog->search engine optimisation->category url suffix)

### Steps to reproduce (*)

1. Choose any category e.g. Men -> Tops -> Tees
2. Select a filter e.g colour red

### Expected result (*)

1. Products are filtered by chosen filter - e.g. just red T-shirts
2. 'Now Shopping By' should show the filtered attribute e.g. red

### Actual result (*)

1. No filters are applied and all category products are shown
2.  'Now Shopping By'  doesnt display any filter

The problem is caused by the modifications to module-url-redirect/Controler/Router.php with the new function getNormalizedPathInfo(RequestInterface $request) that removes trailing slashes, which is fine under default configurations but not if categories are configured to end with a slash

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @Den4ik 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-22731` were created

**Issue Available**: @Den4ik, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by Den4ik** [comment]
```
Request URL: https://23dev.magento2.loc/category-3/?mycolor=5
Request Method: GET
Status Code: 301 
...
location: https://23dev.magento2.loc/category-3/
```

**Comment by NetStorm84** [comment]
This is the PR responsible. It's removing the trailing slash from the path.

https://github.com/magento/magento2/pull/20901

**Comment by NetStorm84** [comment]
This is the commit responsible as it's removing the trailing slash. @serhii-balko 

https://github.com/magento/magento2/commit/16b6a0cb81577d82f8dc6e729c48381fc934530f#diff-2e67977e528b556d3c20ad36dc858f92

I have attached a temporary fix in the form of a .patch file below, if anyone is interested.

[url-rewrite.zip](https://github.com/magento/magento2/files/3808979/url-rewrite.zip)

**Comment by hostep** [comment]
It looks like they are rolling back the change which introduced the problem: https://github.com/magento/magento2/commit/fa468e7298b42c0a233bbcb7db932f8a8ccbb32f

**Comment by NetStorm84** [comment]
> It looks like they are rolling back the change which introduced the problem: [fa468e7](https://github.com/magento/magento2/commit/fa468e7298b42c0a233bbcb7db932f8a8ccbb32f)

Good to know! Thanks for that.

**Comment by pefferen** [comment]
We are experiencing the same issue in 2.2.10

**Comment by robhuijben** [comment]
Same issue happens on 2.2.11

**Comment by engcom-Hotel** [comment]
Hello @wibbel,
Checking the issue:
Magento version: 2.4-develop
PHP version: 7.4
ElasticSearch version: 7
MySQL version: 8.0
1. Changed '.html' to '/' at
![25447_configuration](https://user-images.githubusercontent.com/51681618/106018538-4b6c9680-60ca-11eb-9b86-0cfd7e15c017.png)
2. Opened  'Men -> Tops -> Tees'
3. Filtered by colour
![25447_works_as_expected](https://user-images.githubusercontent.com/51681618/106018673-73f49080-60ca-11eb-90f5-67151c80c564.png)
Works as expected.
Could you please let me know if you are able to reproduce it on 2.4 and if yes - please provide more detailed steps.
Thank you.

**Comment by engcom-Hotel** [comment]
Closing the issue due to inactivity.
Feel free to reopen it if you can reproduce it.

**Comment by magento-engcom-team** [comment]
Hi @wibbel, @engcom-Hotel. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-25117` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
